src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala

```diff
@@ -658,7 +659,10 @@ final case class AgreementApiServiceImpl(
         suspendedByProducer = suspendedByProducer,
         suspendedByPlatform = suspendedByPlatform,
         stamps = agreement.stamps
-          .copy(suspensionByConsumer = suspensionByConsumerStamp, suspensionByProducer = suspensionByProducerStamp)
+          .copy(
+            suspensionByConsumer = if (!suspendedByConsumer.getOrElse(false)) None else suspensionByConsumerStamp,
+            suspensionByProducer = if (!suspendedByProducer.getOrElse(false)) None else suspensionByProducerStamp),
+        suspendedAt = offsetDateTimeSupplier.get().some
```

Comment:
asking just because I'm not aware of the requirement here: if a suspended agreement is suspended again (because for example it was suspended first by producer and then by consumer), this field should be updated with the most recent suspension? Or it indicates since when the agreement was originally suspended?

Reply:
@beetlecrunch As far as I remember we said that here the suspended date should be changed even if double suspension occurs. Can you confirm that?

Reply:
Resolved here by checking if a suspension date is already present. 🆗